homeassistant/components/sms/__init__.py

```diff
@@ -19,15 +21,43 @@
 
 def setup(hass, config):
     """Configure Gammu state machine."""
-    conf = config[DOMAIN]
-    device = conf.get(CONF_DEVICE)
-    gateway = gammu.StateMachine()  # pylint: disable=no-member
-    try:
-        gateway.SetConfig(0, dict(Device=device, Connection="at"))
-        gateway.Init()
-    except gammu.GSMError as exc:  # pylint: disable=no-member
-        _LOGGER.error("Failed to initialize, error %s", exc)
-        return False
-    else:
-        hass.data[DOMAIN] = gateway
+    hass.data.setdefault(DOMAIN, {})
+    sms_config = config.get(DOMAIN, {})
+    if not sms_config:
         return True
+
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=sms_config,
+        )
+    )
+
+    return True
+
+
+async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry):
+    """Configure Gammu state machine."""
+
+    entry_id = entry.entry_id
+    hass.data[DOMAIN].setdefault(entry_id, {})
```

Comment:
Sorry, there was a missing `not` in my question.

Reply:
> Sorry, there was a missing `not` in my question.
No problem, but bear with me here, what are we supposed to store in hass.data[DOMAIN]?
and what is entry.entry_id ? (I copied this code from the tesla powerwall integration)

Reply:
Integrations can store whatever integration specific data they need to share between modules or store for later use in hass.data.

The `entry.entry_id` is a uuid for the config entry. It's common to use this id as a key under `hass.data[DOMAIN]` to key data per that config entry for integrations that allow more than one config entry.

Reply:
> Integrations can store whatever integration specific data they need to share between modules or store for later use in hass.data.
> 
> The `entry.entry_id` is a uuid for the config entry. It's common to use this id as a key under `hass.data[DOMAIN]` to key data per config entry for integrations that allow more than one config entry.

How do I get the entry_id from the notify service in the file notify.py?

Reply:
The notify platform doesn't support config entry setup yet. So we would have to store it ourselves in hass.data. So it may not make sense to use the entry_id.

Reply:
> The notify platform doesn't support config entry setup yet. So we would have to store it ourselves in hass.data. So it may not make sense to use the entry_id.

Can you clarify if SMS still needs to set up an unique ID?

Reply:
> > The notify platform doesn't support config entry setup yet. So we would have to store it ourselves in hass.data. So it may not make sense to use the entry_id.
> 
> Can you clarify if SMS still needs to set up an unique ID?

I think I understand, because the notify service don't support config entries we have to store our data in [DOMAIN][SMS_GATEWAY]. We can change this later when Notify supports config entries (maybe a breaking change?)
For now, it should be keep using [DOMAIN][SMS_GATEWAY]

Reply:
Yes, and then we can remove this line here.